src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1228,7 +1230,7 @@ protected long getApproxSizeOfRecordInBytes(SinkRecord kafkaSinkRecord) {
           }
         }
       }
-    } catch (JsonProcessingException e) {
+    } catch (Exception e) {
       // We ignore any errors here because this is just calculating the record size
```

Comment:
could we at least log sth here?

Reply:
Let's catch `SnowflakeKafkaConnectorException` with `ERROR_0010` code only. similar to what you do in `getData()` below.

Reply:
Good point. Done.